File: web-client/src/app/services/shift-type.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ShiftType } from '../models/shift-type.model';

@Injectable({
  providedIn: 'root'
})
export class ShiftTypeService {
  private baseUrl = '/api/shift-types';

  constructor(private http: HttpClient) { }

  getAllShiftTypes(): Observable<ShiftType[]> {
    return this.http.get<ShiftType[]>(this.baseUrl);
  }

  getShiftTypeById(id: number): Observable<ShiftType> {
    return this.http.get<ShiftType>(`${this.baseUrl}/${id}`);
  }

  createShiftType(shiftType: ShiftType): Observable<ShiftType> {
    return this.http.post<ShiftType>(this.baseUrl, shiftType);
  }

  updateShiftType(shiftType: ShiftType): Observable<ShiftType> {
    return this.http.put<ShiftType>(`${this.baseUrl}/${shiftType.id}`, shiftType);
  }

  deleteShiftType(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  getActiveShiftTypes(): Observable<ShiftType[]> {
    return this.http.get<ShiftType[]>(`${this.baseUrl}/active`);
  }
}

File: api-module/src/main/java/com/shiftmanager/api/service/ShiftTypeService.java
package com.shiftmanager.api.service;

import java.util.List;

import com.shiftmanager.api.dto.ShiftTypeDTO;

public interface ShiftTypeService {
    
    List<ShiftTypeDTO> getAllShiftTypes();
    
    ShiftTypeDTO getShiftTypeById(Long id);
    
    ShiftTypeDTO createShiftType(ShiftTypeDTO shiftTypeDTO);
    
    ShiftTypeDTO updateShiftType(Long id, ShiftTypeDTO shiftTypeDTO);
    
    void deleteShiftType(Long id);
    
    List<ShiftTypeDTO> getActiveShiftTypes();
}
